gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -335,7 +335,9 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
       if (quotaManager.isPresent()) {
         // QuotaManager has idempotent checks for a dagNode, so this check won't double add quotas for a flow in the DagManager
         try {
+          // todo : we should probably check quota for all of the start nodes
```

Comment:
do u mean this for all hops in multi-hop flow?

Reply:
I mean for all the start nodes, not for all the nodes.
So, a Dag has `n` nodes, some of which can be startNodes, that means startNodes can start concurrently. So It makes more sense to check quota on all the start nodes when the ad hoc flow request comes, instead of just checking for it for the node with 0th index.

Reply:
I think quota is at flow level but not job level? If that's the case, checking once should be enough?